### PR TITLE
TYP: Add typing for HDF and synchronize API

### PR DESCRIPTION
**Pull request by bashtage**
- [X] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by bashtage** [comment]
A slightly larger set included than strict documented API. I used my knowledge of interacting with `HDFStore` when deciding what should be in it. Same for `TableIterator`

File: pandas-stubs/io/pytables.pyi [hunk 1]
```diff
@@ -1,76 +1,102 @@
...
+    errors: Literal[
+        "strict",
+        "ignore",
+        "replace",
+        "surrogateescape",
+        "xmlcharrefreplace",
+        "backslashreplace",
+        "namereplace",
+    ]
+    | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/io/pytables.pyi:43]: Not sure that `None` is acceptable for `errors` argument.

File: tests/test_io.py [hunk 2]
```diff
@@ -90,3 +96,39 @@ def test_clipboard_iterator():
         assert_type(read_clipboard(iterator=False, chunksize=1), TextFileReader),
         TextFileReader,
     )
+
+
+def test_hdf():
+    with ensure_clean() as path:
+        check(assert_type(DF.to_hdf(path, "df"), None), type(None))
+        check(assert_type(read_hdf(path), Union[DataFrame, Series, Index]), DataFrame)
```
Comment (Dr-Irv) [inline tests/test_io.py:104]: Should we check `to_hdf()` for a `Series`?  Also, I don't think there is an `Index.to_hdf()`, so is `Index` a possible result of `read_hdf()`?

File: tests/test_io.py [hunk 3]
```diff
@@ -90,3 +96,39 @@ def test_clipboard_iterator():
...
+
+
+def test_hdfstore():
+    with ensure_clean() as path:
+        store = HDFStore(path, model="w")
```
Comment (twoertwein) [inline tests/test_io.py:109]: might be good using it as a context manager - tests also whether `__enter__` and `__exit__` are declared.

File: tests/test_io.py (hunk 2 above)
Comment (bashtage) [reply tests/test_io.py:110]: I thought one could store indices but nope. Remvoed.

File: tests/test_io.py (hunk 3 above)
Comment (bashtage) [reply tests/test_io.py:115]: Added a test

File: pandas-stubs/io/pytables.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/io/pytables.pyi]: Fixed.

**Review by Dr-Irv — COMMENTED** [review]
Can you add a test for writing/reading a `Series` ?

**Comment by bashtage** [comment]
> Can you add a test for writing/reading a `Series` ?

Added one.

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Pull request merged**